src/test/java/com/alkemy/ong/mapper/CategoryMapperTest.java

```diff
@@ -0,0 +1,153 @@
+package com.alkemy.ong.mapper;
+
+import com.alkemy.ong.dto.CategoryDto;
+import com.alkemy.ong.dto.CategoryDtoFull;
+import com.alkemy.ong.entity.CategoryEntity;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mapstruct.factory.Mappers;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class CategoryMapperTest {
+    private static CategoryMapper mapper;
+
+    @BeforeAll
+    public static void setUp() {
+        mapper= Mappers.getMapper(CategoryMapper.class);
+
+    }
+    @Test
+    void categoryDtoToCategoryEntity() {
+        CategoryDto dto= new CategoryDto();
+        dto.setImage("IMAGE ");
+        dto.setName("NAME category");
+        dto.setDescription("Description123");
+        CategoryEntity ent= mapper.categoryDtoToCategoryEntity(dto);
+        assertAll(
+                ()->{
+                    assertEquals(dto.getName(),ent.getName());
+                    assertEquals(dto.getImage(), ent.getImage());
+                    assertEquals("Description123",ent.getDescription());
+                }
+        );
+    }
+    @Test
+    public void categoryDtoMapperTestSimpleNull(){
+        CategoryDto dto=null;
+        CategoryEntity entity=mapper.categoryDtoToCategoryEntity(dto);
+        assertEquals(null,entity);
+    }
+
+    @Test
+    void categoryEntityToCategoryDto() {
+        CategoryEntity category= new CategoryEntity();
+        category.setId("1234");
+        category.setImage("Imaggen category");
+        category.setName("nombre category");
+        category.setSoftDelete(false);
+        category.setDescription("Description");
+        category.setTimestamps(LocalDateTime.now());
+
+        CategoryDto dto=mapper.categoryEntityToCategoryDto(category);
+        assertAll(
+                ()->{
+                    assertEquals(dto.getName(),category.getName());
+                    assertEquals(dto.getImage(), category.getImage());
+                    assertEquals("Description",category.getDescription());
+                }
+        );
+    }
+    @Test
+    public void categoryMapperTestSimpleNull(){
+        CategoryEntity entity=null;
+        CategoryDto dto=mapper.categoryEntityToCategoryDto(entity);
+        assertEquals(null,dto);
+    }
+    @Test
+    public void categoryListMapperTestSimpleNull(){
+        List<CategoryEntity> categoryEntityList=null;
+        List<CategoryDto>categoryDtoList=mapper.listCategoryEntityToListCategoryDto(categoryEntityList);
+        assertEquals(null,categoryDtoList);
+    }
+
+    @Test
+    void listCategoryEntityToListCategoryDto() {
+
+        CategoryEntity categoryL= new CategoryEntity();
```

Comment:
I suggest to implement a Mock class in the util package to reuse and make the code easier to read

Reply:
Thanks for watching, we don't have time for it anymore.

Reply:
I understand